core/grant/service.go

```diff
@@ -673,3 +674,7 @@ func getGrantIDs(grants []domain.Grant) []string {
 func (s *Service) GetGrantsTotalCount(ctx context.Context, filters domain.ListGrantsFilter) (int64, error) {
 	return s.repo.GetGrantsTotalCount(ctx, filters)
 }
+
+func (s *Service) ListUserRoles(ctx context.Context, filters domain.ListGrantsFilter) ([]string, error) {
```

Comment:
let's create a new `ListUserRolesFilter` struct instead of using `ListGrantsFilter` here